common/common_test.go

```diff
@@ -877,6 +877,148 @@ func TestGetOrCreateLastExecutionResultForPrevHeader(t *testing.T) {
 	})
 }
 
+func Test_ExtractBaseExecutionResultHandler(t *testing.T) {
+	t.Parallel()
+
+	t.Run("in case of nil lastExecResultsHandler should return ErrNilLastExecutionResultHandler", func(t *testing.T) {
+		t.Parallel()
+
+		baseExecRes, err := common.ExtractBaseExecutionResultHandler(nil)
+		require.Nil(t, baseExecRes)
+		require.Equal(t, common.ErrNilLastExecutionResultHandler, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		expectedBaseExecResult := &block.BaseMetaExecutionResult{}
+		baseExecRes, err := common.ExtractBaseExecutionResultHandler(&block.MetaExecutionResultInfo{
+			ExecutionResult: expectedBaseExecResult,
+		})
+		require.Nil(t, err)
+		require.Equal(t, expectedBaseExecResult, baseExecRes)
+	})
+
+	t.Run("in case of nil ExecutionResult on MetaExecutionResultInfo should return ErrNilBaseExecutionResult", func(t *testing.T) {
+		t.Parallel()
+
+		baseExecRes, err := common.ExtractBaseExecutionResultHandler(&block.MetaExecutionResultInfo{
+			ExecutionResult: nil,
+		})
+		require.Nil(t, baseExecRes)
+		require.Equal(t, common.ErrNilBaseExecutionResult, err)
+	})
+
+	t.Run("in case of wrong base execution result should return unsupported execution result handler type", func(t *testing.T) {
+		t.Parallel()
+
+		expectedBaseExecResult := &block.BaseExecutionResult{}
+		baseExecRes, err := common.ExtractBaseExecutionResultHandler(&block.ExecutionResult{
+			BaseExecutionResult: expectedBaseExecResult,
+		})
+		require.ErrorContains(t, err, "unsupported execution result handler type")
+		require.Nil(t, baseExecRes)
+	})
+
+	t.Run("should work in case of ExecutionResultInfo type", func(t *testing.T) {
+		t.Parallel()
+
+		expectedBaseExecResult := &block.BaseExecutionResult{}
+		baseExecRes, err := common.ExtractBaseExecutionResultHandler(&block.ExecutionResultInfo{
+			ExecutionResult: expectedBaseExecResult,
+		})
+		require.Nil(t, err)
+		require.Equal(t, expectedBaseExecResult, baseExecRes)
+	})
+
+	t.Run("should return ErrNilBaseExecutionResult in case of nil ExecutionResult on ExecutionResultInfo", func(t *testing.T) {
+		t.Parallel()
+
+		_, err := common.ExtractBaseExecutionResultHandler(&block.ExecutionResultInfo{
+			ExecutionResult: nil,
+		})
+		require.Equal(t, common.ErrNilBaseExecutionResult, err)
+	})
+}
+
+func Test_GetOrCreateLastExecutionResultForPrevHeader(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work in case of headerV3", func(t *testing.T) {
+		t.Parallel()
+
+		baseExecResult := &block.BaseExecutionResult{}
```

Comment:
can you add some relevant data?
at least to not be the default initialization, as we check for equality afterwards